File: 2016_10_29/bubble_sort/bubble_sort.js
#!/usr/bin/node

function bubble_sort(a) {
	for (let x in a)
		for (let y = 1; y <= a.length - x; y++)
			if (a[y] < a[y-1])
				[a[y], a[y-1]] = [a[y-1], a[y]];
}





// Test
let test = [];
for (let i = 0; i < 20; i++)
	test.push(i);

for (let i = 0; i < 20; i++) {
	let n = Math.floor(Math.random() * (20 - i) + i);
	[test[i], test[n]] = [test[n], test[i]];
}
console.log(test);
bubble_sort(test);
console.log(test);

File: 2016_10_29/bubble_sort/bubble_sort.lua
#!/usr/bin/lua

function bubble_sort(a)
	for x,v in ipairs(a) do
		for y = 2,(#a - x + 1) do
			if a[y] < a[y-1] then
				a[y],a[y-1] = a[y-1],a[y]
			end
		end
	end
end

local test = {};
for i = 1,20 do
	table.insert(test, i)
end

math.randomseed(os.time())
for i = 1,19 do
	local n = math.random(i+1, 20)
	test[i],test[n] = test[n],test[i]
end

print(table.concat(test, " "))
bubble_sort(test)
print(table.concat(test, " "))

